### PR TITLE
net: buf: remove use of special putter and getter functions

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Convert users of `net_buf_put()` and `net_buf_get()` functions to use non-wrapped putters and getters (`k_fifo_put()`, `k_fifo_get()`) and deprecate the wrappers.

Special handling of net_bufs in k_fifos is no longer needed after commit 3d306c1, since these actions are now atomic regardless of any net_buf fragments.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

### CODE REVIEW

File: subsys/bluetooth/host/classic/rfcomm.c [hunk 1]
```diff
@@ -1748,7 +1748,7 @@ int bt_rfcomm_dlc_disconnect(struct bt_rfcomm_dlc *dlc)
 		 * and stop the tx thread.
 		 */
 		dlc->state = BT_RFCOMM_STATE_USER_DISCONNECT;
-		net_buf_put(&dlc->tx_queue,
+		k_fifo_put(&dlc->tx_queue,
```
Comment (jhedberg) [inline subsys/bluetooth/host/classic/rfcomm.c:1751]: minor: there's a clang-format suggestion to put the call on a single line. Would also fix the misalignment of the split line that's caused by the new function being one character shorter.

**Review by jhedberg — DISMISSED** [review]
Nice, thanks! Just a pity that we didn't understand to do this earlier, which would have let us remove the old API sooner.

**Review by nordicjm — DISMISSED** [review]
MCUmgr changes OK

**Review by tmon-nordic — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by nordicjm — DISMISSED** [review]
MCUmgr changes OK

**Review by jukkar — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: doc/connectivity/networking/api/net_buf.rst [hunk 2]
```diff
@@ -49,15 +49,8 @@ buffers may also contain protocol-specific context, known as user data.
...
 
-The buffers have native support for being passed through k_fifo kernel
-objects. This is a very practical feature when the buffers need to be
-passed from one thread to another. However, since a net_buf may have a
-fragment chain attached to it, instead of using the :c:func:`k_fifo_put`
-and :c:func:`k_fifo_get` APIs, special :c:func:`net_buf_put` and
-:c:func:`net_buf_get` APIs must be used when passing buffers through
-FIFOs. These APIs ensure that the buffer chains stay intact. The same
-applies for passing buffers through a singly linked list, in which case
-the :c:func:`net_buf_slist_put` and :c:func:`net_buf_slist_get`
```
[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply doc/connectivity/networking/api/net_buf.rst:60]: Please point to which code lines implement "native support for being passed through k_fifo kernel objects"?

[the remark this answers is not in the document]
Comment (jfischer-no) [reply doc/connectivity/networking/api/net_buf.rst:60]: https://github.com/zephyrproject-rtos/zephyr/blob/50a23a61950fa88fc9baf913421338b336fe4b96/include/zephyr/net/buf.h#L1007-L1008

[the remark this answers is not in the document]
Comment (henrikbrixandersen) [reply doc/connectivity/networking/api/net_buf.rst]: Thanks, added.

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]
MCUmgr parts OK

**Pull request merged**